content-test/addon/MetadataParser.test.js

```diff
@@ -0,0 +1,72 @@
+/* globals assert, require */
+"use strict";
+const createMetadataParser = require("inject!addon/MetadataParser");
+const TEST_HTML = `<html>
+  <head>
+    <title>Test Page</title>
+    <link rel="shortcut icon" type="image/x-icon" href="//www.foo.com/icon.png" />
+    <meta property="og:image" content="//www.foo.com/image.png" />
+  </head>
+ </html>`;
+
+describe("MetadataParser", () => {
+  let metadataParser;
+  function setup() {
+    const mockPreviewProvider = {
+      processLinks(link) {
+        return link.map(link => Object.assign({}, link, {cache_key: link.url, places_url: link.url}));
+      }
+    };
+    const {MetadataParser} = createMetadataParser({
+      "common/vendor.bundle": {
+        PageMetadataParser: {getMetadata(doc) {return doc;}},
```

Comment:
The return signature of getMetadata is not a document, it's an object with the following format:

https://github.com/mozilla/page-metadata-parser#usage

But you should only mock this out if you have to force its behaviour in some way.  If you're just testing the success case, just actually invoke it.  It's synchronous, stateless, and self contained, so you lose nothing by actually calling it.